src/utils.c

```diff
@@ -1051,3 +1042,24 @@ ssize_t read_size(const int fd, void *buf, const size_t size) {
 
     return got;
 }
+
+/*
+ * Get a directory file descriptor from a given DIR *, infallibly.
+ *
+ * This is just a wrapper around libc's dirfd() that panics if an error is reported, since
+ * an error in dirfd() always indicates a bug in GUFI.
+ */
+int gufi_dirfd(DIR *d) {
+    int d_fd = dirfd(d);
+    if (d_fd < 0) {
+        /*
+         * We should never get here. glibc's dirfd(3) never return errors, and
+         * Apple's libc only returns an error if the DIR * is invalid, which would
+         * indicate a bug in GUFI.
+         */
+        fprintf(stderr, "BUG: dirfd(3) failed: errno = %d\n", errno);
+        exit(1);
+    }
```

Comment:
Leave the comment and remove the check?

Reply:
I could go either way on this. This check does seem kind of stupid in a way, but on the other hand being explicit about errors does seem like "good hygiene".

Probably the only realistic reason to hit this would be accidentally forgetting to check the return of `opendir()` and passing a NULL pointer to `dirfd()`. On Linux using glibc, this would immediately segfault, but on Mac, this would return EINVAL.

In that case, the program would shortly crash anyways as most likely `opendir()` will be followed by `readdir()` which will dereference the NULL `DIR *`.

So basically, if this error is ignored, the program is gonna crash either way. Its just whether it segfaults, or if it prints out an error message and then exits.